domain/src/main/java/org/akvo/flow/domain/interactor/CreateUser.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.domain.interactor;
+
+import org.akvo.flow.domain.executor.PostExecutionThread;
+import org.akvo.flow.domain.executor.ThreadExecutor;
+import org.akvo.flow.domain.repository.SurveyRepository;
+import org.akvo.flow.domain.repository.UserRepository;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observable;
+import io.reactivex.functions.Function;
+
+public class CreateUser extends UseCase {
+
+    public static final String PARAM_USER_NAME = "user";
+    private static final long INVALID_ID = -1L;
```

Comment:
Do you see a disadvantage using `SharedPreferencesDataSource.INVALID_ID`?  In the event of a decision to change what this number is, we can keep it consistent..

Reply:
CreateUser does not know about SharedPreferenceDataSource and I want to keep them separated. So here I renamed the contsant to INVALID_USER_ID. I do not see any reaosn for the value to change as -1 is the value returned by the database when insert has failed. On the other hand INVALID_ID in SharedPreferenceDataSource can represent any value which is missing and is a long.